latch/resources/workflow.py

```diff
@@ -1,95 +1,94 @@
     if isinstance(metadata, Callable):
-        return _workflow(metadata)
-    else:
-
-        def decorator(f: Callable):
-            if f.__doc__ is None:
-                f.__doc__ = f"{f.__name__}\n\nSample Description"
-            short_desc, long_desc = f.__doc__.split("\n", 1)
-
-            signature = inspect.signature(f)
-            wf_params = signature.parameters
-
-            in_meta_not_in_wf = []
-            not_in_meta_in_wf = []
-
-            for meta_param in metadata.parameters:
-                if meta_param not in wf_params:
-                    in_meta_not_in_wf.append(meta_param)
-
-            for wf_param in wf_params:
-                if wf_param not in metadata.parameters:
-                    not_in_meta_in_wf.append(wf_param)
-
-            if len(in_meta_not_in_wf) > 0 or len(not_in_meta_in_wf) > 0:
-                error_str = (
-                    "Inconsistency detected between parameters in your `LatchMetadata`"
-                    " object and parameters in your workflow signature.\n\n"
-                )
-
-                if len(in_meta_not_in_wf) > 0:
-                    error_str += (
-                        "The following parameters appear in your `LatchMetadata` object"
-                        " but not in your workflow signature:\n\n"
-                    )
-                    for meta_param in in_meta_not_in_wf:
-                        error_str += f"    \x1b[1m{meta_param}\x1b[22m\n"
-                    error_str += "\n"
-
-                if len(not_in_meta_in_wf) > 0:
-                    error_str += (
-                        "The following parameters appear in your workflow signature but"
-                        " not in your `LatchMetadata` object:\n\n"
-                    )
-                    for meta_param in not_in_meta_in_wf:
-                        error_str += f"    \x1b[1m{meta_param}\x1b[22m\n"
-                    error_str += "\n"
-
-                error_str += (
-                    "Please resolve these inconsistencies and ensure that your"
-                    " `LatchMetadata` object and workflow signature have the same"
-                    " parameters."
-                )
-
-                raise ValueError(error_str)
-
-            for name, meta_param in metadata.parameters.items():
-                if meta_param.samplesheet is not True:
-                    continue
-
-                annotation = wf_params[name].annotation
-
-                origin = get_origin(annotation)
-                args = get_args(annotation)
-                valid = (
-                    origin is not None
-                    and issubclass(origin, list)
-                    and is_dataclass(args[0])
+        f = metadata
+        signature = inspect.signature(f)
+        wf_params = signature.parameters
+        metadata = LatchMetadata(f.__name__, LatchAuthor())
+        metadata.parameters = _generate_params(wf_params)
+
+        _inject_metadata(f, metadata)
```

Comment:
this is a breaking change as it will break really old workflows (prior to July 2022) where the parameter metadata was contained in the docstring as YAML

we should maintain backwards compatibility for now, and add this to the list of things to drop when we finally release 3.0.0